### ISSUE TITLE
auth with client credentials

### ISSUE THREAD

**Issue by christw92** [sidorares/node-mysql2#352]
I think there should be a possibility to: not provide username & password in config.js and instead require the user to provide user & pw when connecting to the node-server (which will then be used for executing queries on the mysql-server)

**Comment by sidorares** [comment]
you mean ask at handshake time? How it is better compared to 1) ask user for credentials 2) create connection?

**Comment by christw92** [comment]
Yes I mean at handshake time (that word did not came to my mind when I opened the issue).
I don't think one option is better than the other one. I think it would be great if both 1) and 2) would be supported. It depends on the use case.

e.g.: I'm thinking about implementing a "mysql-proxy" that automatically handles data-historization (auto-copy data that is going to be modified/deleted to other tables before actually performing the UPDATE/DELETE).

For this kind of things it would be great to ask at handshake time because in this case there is no other time when the user/database-application can be asked for credentials

**Comment by sidorares** [comment]
It'll be possible when custom connection time AuthPlugin is added - see https://github.com/sidorares/node-mysql2#authentication-switch-request

On the server there is `authCallback` parameter  https://github.com/sidorares/node-mysql2/blob/aff0b1c8eca5c63566786c72d645224d5a914170/lib/commands/server_handshake.js#L44
